src/app/shared/pipes/logActionDisplay.ts

```diff
@@ -0,0 +1,20 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+function formatLogAction (type: 'result_started'|'submission'|'result_validated', score?: number): string {
+  if (type === 'submission' && score === undefined) {
+    return $localize`Submission`;
+  } else if (type === 'submission' && score !== undefined) {
+    return $localize`Submission (score: ${ score })`;
+  } else if (type === 'result_started') {
+    return $localize`Activity started`;
+  } else if (type === 'result_validated') {
+    return $localize`Activity validated`;
+  } else {
+    return type;
+  }
+}
+
+@Pipe({ name: 'logActionDisplay' })
```

Comment:
👍 
I would say that pipe is very specific to this component only, so I would put it in the "item-log-view" component directory.

Reply:
fixed